Training/ClassLibrary/ClassLib.cs

```diff
@@ -1,5 +1,56 @@
-﻿namespace ClassLibrary {
-   public class Class1 {
+﻿// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joinees at Metamation, Batch - July 2023.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------------
+// Classlib.cs
+// Program that parses file paths and extracts data, including the file name, extension,
+// drive letter, and folder.
+// ------------------------------------------------------------------------------------------------
+using static State;
+/// <summary>Provides method for interpreting file paths and retrieving data, including 
+/// file name, extension, drive letter, and folder </summary>
+#region FileNameParser -------------------------------------------
+public class FileNameParser {
+   #region Methods -----------------------------------------------
+   /// <summary>Represents the various stages that occur 
+   /// during the parsing procedure </summary>
+   /// See File://
+   public static (string driveLetter, string folder, string fileName, string fileExtn) ParseFileName (string input) {
+      string driveLetter = string.Empty, folder = string.Empty, fileName = string.Empty, fileExtn = string.Empty, temp = string.Empty;
+      State currentState = A;
+      Action none = () => { };
+      Action toDo;
+      string alphabets = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

```

Comment:
Why use this when you had already used value comparison in line 26? Both alphabets.Contains and ">= 'A' and <= 'Z'" check the same condition.